app/models/state_file_base_intake.rb

```diff
@@ -132,8 +132,22 @@ def spouse_deceased?
     direct_file_data.spouse_deceased?
   end
 
-  def self.locality_nm_valid?(locality)
-    true
+  def validate_w2(w2)
+  end
+
+  def invalid_df_w2?(df_w2)
```

Comment:
sorry if this is a bit tedious but i think that if all this code moved here, the tests should also be moved to the model spec. seems strange to leave it in the controller specs.

Reply:
I was on the fence about this - but I agree. Done